### ISSUE TITLE
Add instrumentation for JAX-RS/Jersey Client

### ISSUE THREAD

**Issue by masooh** [spring-cloud/spring-cloud-sleuth#726]
It would be great to have also support for JAX-RS/Jersey clients. The implementation of Feign and RestTemplate can be used as guidance. Required elements:

- `JaxrsTraceFilter` implementing `ClientRequestFilter` and `ClientResponseFilter`. The Filter starts and stops the span, logs the start and stop events, all like the `TraceRestTemplateInterceptor`.
- `JaxrsClientAutoConfiguration` like `TraceWebClientAutoConfiguration` creates the `JaxrsTraceFilter` and registers it on existing JAX-RS clients. The class condition is on `javax.ws.rs.client.Client`. This class has to be added to the `EnableAutoConfiguration` entry in `spring.factories`.

I would provide a pull request in the near future. Feedback to the implementation plan above is welcome.

**Comment by marcingrzejszczak** [comment]
Does this also support the exception handling? I think you wrote sth about this on Gitter but maybe I'm wrong?

**Comment by masooh** [comment]
@marcingrzejszczak Sorry, didn't see your comment.

You understood me right on Gitter. Unfortunately it does not support exception handling. I tried an attempt with Spring AOP. But in this case I would have to create 3 Proxies, 2 of them just to create another Proxy so that I finally can wrap `JerseyInvocation#invoke()`.
That looked not good for me, therefore I created a issue for Jersey: [Add possibility to register exception handler to client ](https://github.com/jersey/jersey/issues/3709).

Therefore this ticket can be closed and reopened if the Jersey ticket is fullfilled.

For the sake of documentation: My first attempt without exception handling is here: https://github.com/masooh/spring-cloud-sleuth/commits/jersey-client-instrumentation
